vesoft-inc/nebula
# some fix for third-part build

## What I want to do
build nebula graph on ArchLinux, I need to rebuild all third-party and create a new nebula build toolchain.

### Why are the changes needed?
glibc 2.27 is too old #2333 

this PR fix third-party building on glibc 2.32 / gcc 10 / kernel 5.8.5 / cmake 3.18

### Will break the compatibility? How if so?
almost in third-party cmake file, not affect existing code.

### Does this PR introduce any user-facing change?
no.

### How was this patch tested?
there is two major change:
  - add `-fcommon` when using GCC 10
  - add `Boost_NO_SYSTEM_PATHS` to make sure using vendor boost.

the first problem is very easy to reproduce: add `-fno-common` to CFLAGS, rebuild third-party using GCC 9. krb5 and libunwind get an error when linking.

the second problem: install a new Ubuntu 19.04. run `bash third-party/build-third-party.sh` got this warning
    
    New Boost version may have incorrect or missing dependencies and imported targets.

and after check the link script, when building fbthrift and folly, LD will link system default boost, not the vendor boost. if install a new boost on Ubuntu 19.04 will get link error.

the last problem is just a warning about NCURSES library naming.

### Checklist
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above #2333
- [ ] I've informed the technical writer about the documentation change if necessary
